docs/how-to-guides/deploy-a-flow/deploy-using-kubernetes.md

```diff
@@ -2,3 +2,194 @@
+3. Create Kubernetes deployment yaml.
+4. Apply the deployment.
+
+
+## Build a flow as docker format
+
+::::{tab-set}
+:::{tab-item} CLI
+:sync: CLI
+
+Note that all dependent connections must be created before building as docker.
+```bash
+# create connection if not created before
+pf connection create --file ../../../examples/connections/azure_openai.yml --set api_key=<your_api_key> api_base=<your_api_base> --name open_ai_connection
+```
+
+Use the command below to build a flow as docker format:
+```bash
+pf flow build --source <path-to-your-flow-folder> --output <your-output-dir> --format docker
+```
+:::
+:::{tab-item} VS Code Extension
+:sync: VSC
+
+Click the button below to build a flow as docker format:
+![img](../../media/how-to-guides/vscode_export_as_docker.png)
+:::
+::::
+
+Note that all dependent connections must be created before exporting as docker.
+
+### Docker format folder structure
+
+Exported Dockerfile & its dependencies are located in the same folder. The structure is as below:
+- flow: the folder contains all the flow files
+  - ...
+- connections: the folder contains yaml files to create all related connections
+  - ...
+- Dockerfile: the dockerfile to build the image
+- start.sh: the script used in `CMD` of `Dockerfile` to start the service
+- settings.json: a json file to store the settings of the docker image
+- README.md: Simple introduction of the files
+
+## Deploy with Kubernetes
+We are going to use the [web-classification](https://github.com/microsoft/promptflow/tree/main/examples/flows/standard/web-classification/) as
+an example to show how to deploy with Kubernetes.
+
+Please ensure you have [create the connection](../manage-connections.md#create-a-connection) required by flow, if not, you could
+refer to [Setup connection for web-classifiction](https://github.com/microsoft/promptflow/tree/main/examples/flows/standard/web-classification).
+
+Additionally, please ensure that you have installed all the required dependencies. You can refer to the "Prerequisites" section in the README of the [web-classification](https://github.com/microsoft/promptflow/tree/main/examples/flows/standard/web-classification/) for a comprehensive list of prerequisites and installation instructions.
+
+### Build Docker image
+
+Like other Dockerfile, you need to build the image first. You can tag the image with any name you want. In this example, we use `web-classification-serve`.
+
+Then run the command below:
+
+```bash
+cd <your-output-dir>
+docker build . -t web-classification-serve
+```
+
+### Create Kubernetes deployment yaml.
+The Kubernetes deployment yaml file serves as a blueprint for orchestrating your docker container within a Kubernetes pod. It meticulously outlines essential details, including the container image, port configurations, environment variables, and various settings. Presented below is a basic deployment template for your convenience, which you can effortlessly tailor to your specific requirements.
```

Comment:
avoid using complicated words, and clause sentences. Maybe chatgpt could resolve this piece in a more user-friendly way.

Reply:
updated